central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -0,0 +1,244 @@
+	Cluster          int
+	ComplianceStatus string
+}
+
+type formatSubject struct {
+	BrandedPrefix string
+	ScanConfig    string
+	Profiles      string
+}
+
+type complianceReportGeneratorImpl struct {
+	checkResultsDS        checkResults.DataStore
+	notificationProcessor notifier.Processor
+}
+
+// struct which hold all columns of a row
+type resultRow struct {
+	ClusterName string
+	CheckName   string
+	Profile     string
+	ControlRef  string
+	Description string
+	Status      string
+	Remediation string
+}
+
+type resultEmail struct {
+	resultCSVs map[string][]*resultRow //map of cluster id to slice of *resultRow
+	totalPass  int
+	totalFail  int
+	totalMixed int
+	profiles   []string
+	clusters   int
+}
+
+func (rg *complianceReportGeneratorImpl) ProcessReportRequest(ctx context.Context, req *ComplianceReportRequest) error {
+
+	data, err := rg.getDataforReport(req, ctx)
+	if err != nil {
+		return err
+	}
+
+	zipData, err := rg.Format(data.resultCSVs)
+	if err != nil {
+		return err
+	}
+	profiles := strings.Join(req.profiles, ", ")
+	formatEmailBody := &formatBody{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		Profile:       profiles,
+		Pass:          data.totalPass,
+		Fail:          data.totalFail,
+		Mixed:         data.totalMixed,
+		Cluster:       len(req.clusterIDs),
+	}
+	formatEmailBody.ComplianceStatus = "non compliant"
+	if data.totalFail == 0 {
+		formatEmailBody.ComplianceStatus = "compliant"
+	}
+
+	if len(profiles) > maxNumberProfilesinSubject {
+		profiles = strings.Join(req.profiles[0:maxNumberProfilesinSubject], ", ")
+		profiles += "..."
+	}
+
+	formatEmailSub := &formatSubject{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		ScanConfig:    req.scanConfigName,
+		Profiles:      profiles,
+	}
+
+	return rg.sendEmail(zipData, formatEmailBody, formatEmailSub, req.notifiers, ctx)
+}
+
+// getDataforReport returns map of cluster id and
+func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportRequest, ctx context.Context) (*resultEmail, error) {
+	// TODO ROX-24356: Implement query to get checkresults data to generate cvs for compliance reporting
+
+	return nil, nil
+}
+
+func (rg *complianceReportGeneratorImpl) sendEmail(zipData *bytes.Buffer, emailBody *formatBody, formatEmailSub *formatSubject, notifiersList []*storage.NotifierConfiguration, ctx context.Context) error {
+
+	errorList := errorhelpers.NewErrorList("Error sending compliance report email notifications")
+	for _, notifier := range notifiersList {
+		nf := rg.notificationProcessor.GetNotifier(ctx, notifier.GetEmailConfig().GetNotifierId())
+		reportNotifier, ok := nf.(notifiers.ReportNotifier)
+		if !ok {
+			errorList.AddError(errors.Errorf("incorrect type of notifier '%s' for compliance report", notifier.GetEmailConfig().GetNotifierId()))
+			continue
+		}
+
+		customBody := notifier.GetEmailConfig().GetCustomBody()
+		emailBody, err := formatEmailBodywithDetails(defaultEmailBodyTemplate, emailBody)
```

Comment:
We can do this once, all data collection and formatting. Then loop through notifiers, and invoke the send and continue. and return from this func.

Reply:
There can be a different email template for each notifier. hence  need to format subject and body each time